interface/forms/group_attendance/functions.php

```diff
@@ -0,0 +1,113 @@
+}
+
+function insert_new_participant($form_id, $new_participant_id, $new_comment){
+    $sql_for_table_tgpa = "INSERT INTO therapy_groups_participant_attendance (form_id, pid, meeting_patient_comment, meeting_patient_status) " .
+        "VALUES(?,?,?,?);";
+    sqlInsert($sql_for_table_tgpa, array($form_id, $new_participant_id, $new_comment, 20));
+}
+
+function insert_patient_appt($pid, $gid, $pc_aid, $pc_eventDate, $pc_startTime, $participantData){
+    $select_sql = "SELECT pc_eid FROM openemr_postcalendar_events WHERE pc_pid = ? AND pc_gid = ? AND pc_eventDate = ? AND pc_startTime = ?;";
+    $result = sqlStatement($select_sql, array($pid, $gid, $pc_eventDate, $pc_startTime));
+    $result_array = sqlFetchArray($result);
+    $converted_status = convertStatus($participantData['status']);
+    if($result_array){
+        $insert_sql = "UPDATE openemr_postcalendar_events SET pc_apptstatus = ? WHERE pc_eid = ?;";
+        sqlInsert($insert_sql, array($converted_status, $result_array['pc_eid']));
+    }
+    else{
+        $insert_sql =
+            "INSERT INTO openemr_postcalendar_events " .
+            "(pc_catid, pc_aid, pc_pid, pc_gid, pc_title, pc_informant, pc_eventDate, pc_recurrspec, pc_startTime, pc_sharing, pc_apptstatus) ".
+            "VALUES (1000, ?, ?, ?, 'Group Therapy', 1, ?, ?, ?, 0, ?); ";
+        $recurrspec = 'a:6:{s:17:"event_repeat_freq";s:1:"0";s:22:"event_repeat_freq_type";s:1:"0";s:19:"event_repeat_on_num";s:1:"1";s:19:"event_repeat_on_day";s:1:"0";s:20:"event_repeat_on_freq";s:1:"0";s:6:"exdate";s:0:"";}';
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $pc_aid, $pid, $gid, $pc_eventDate, $recurrspec, $pc_startTime, $converted_status);
+        sqlInsert($insert_sql, $sqlBindArray);
+    }
+}
+
+function insert_patient_encounter($pid, $gid, $group_encounter_date, $participantData, $pc_aid){
+    $select_sql = "SELECT id FROM form_encounter WHERE pid = ? AND external_id = ? AND pc_catid = ? AND date = ?; ";
+    $result = sqlStatement($select_sql, array($pid, $gid, 1000, $group_encounter_date));
+    $result_array = sqlFetchArray($result);
+    if($result_array){
+        $insert_sql = "UPDATE form_encounter SET reason = ? WHERE id = ?;";
+        sqlInsert($insert_sql, array($participantData['comment'], $result_array['pc_eid']));
+    }
+    else{
+        $insert_encounter_sql =
+            "INSERT INTO form_encounter (date, reason, pid, encounter, pc_catid, provider_id, external_id) ".
+            "VALUES (?, ?, ?, ?, 1000, ?, ?);";
+        $enc_id = generate_id();
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $group_encounter_date, $participantData['comment'], $pid, $enc_id, $pc_aid, $gid);
+        $form_id = sqlInsert($insert_encounter_sql, $sqlBindArray);
+
+        global $userauthorized;
+
+        addForm($enc_id, "New Patient Encounter", $form_id, "newpatient",$pid, $userauthorized, $group_encounter_date,'','',NULL);
+
+    }
+}
```

Comment:
Actually, I fund where you are using above functions below. I'll provide guidance there.